keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/annealing/spring/ApplicationEventResourceCheckListener.kt

```diff
@@ -0,0 +1,21 @@
+package com.netflix.spinnaker.keel.annealing.spring
+
+import com.netflix.spinnaker.keel.annealing.ResourceActuator
+import org.slf4j.LoggerFactory
+import org.springframework.context.event.EventListener
+import org.springframework.scheduling.annotation.Async
+
+open class ApplicationEventResourceCheckListener(
+  private val resourceActuator: ResourceActuator
+) {
+  @Async
+  @EventListener(ResourceCheckEvent::class)
```

Comment:
This is pretty neat because it means that even though the event dispatch is synchronous, this handler runs async. In other words it won't block that loop in `ResourceCheckScheduler`. Instead this work will get run on a thread pool.